src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparamorder/InputJavadocParamOrderIncorrect.java

```diff
@@ -0,0 +1,115 @@
+package com.puppycrawl.tools.checkstyle.checks.javadoc.javadocparamorder;
+
+/**
+ * Config: default
+ * Some Javadoc.
+ *
+ * @since something
+ */
+class InputJavadocParamOrderIncorrect {
+
+    /**
+     * Description.
+     *
+     * @version Some javadoc. // OK
+     * @param p2
+     *            Parameter 2 desc. // violation
```

Comment:
`violation` comments should be on the corresponding line.

Reply:
done

Reply:
Please do not resolve items until I comment "resolved". You have `violation` comments on lines that should never be a violation, such as 
https://github.com/checkstyle/checkstyle/blob/2ab25be6b5002d536e8fb52297e87ede719c3f48/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparamorder/InputJavadocParamOrderDefs.java#L63

For now, ignore what the check is doing, and make correct test inputs for the expected check behavior.  This will give you a more clear idea of how to develop the check.

Reply:
i thought we write ok or violation comment after the left curly according to whether it's producing any tests violation, like the u mentioned above in that we have 3 tests violation so i wrote // violation after left curly.
i was wrong, i will correct this.

Reply:
check now please

Reply:
This will only cause confusion in the future. `// violation` comments should only be on the exact line that appears when running checkstyle over the input with given configuration. Once you correct this and add https://github.com/checkstyle/checkstyle/pull/9699#discussion_r625076948 and https://github.com/checkstyle/checkstyle/pull/9699#discussion_r625078507, I think we will be good to resume development of the check, since we have a clear idea of expected violations.

Reply:
Also remove all `// ok` comments that aren't on lines that would be reported by this check if they were violations. These test inputs should have no `// ok` and `// violation` comments except in Javadocs comment blocks on param tags. That's it.

Reply:
i have removed (in my ide) all the // ok comments in xdoc but i cannt find any // ok and // violation besides that could u point if u found any?

Reply:
https://github.com/checkstyle/checkstyle/blob/a48a97364805835e9dd33e0383d39fa1b520c535/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparamorder/InputJavadocParamOrderDefs.java#L69

There are 43 occurrences like this. These comments do not make sense, especially where we have violations in the associated javadoc. This is a javadoc check, so there should never be any violations anywhere except within the content of a javadoc comment.

Reply:
Resolved.